tests/h/resources_test.py

```diff
@@ -71,30 +78,36 @@ def links_service(self, pyramid_config):
         pyramid_config.register_service(service, name='links')
         return service
 
+    @pytest.fixture
+    def user_service(self, pyramid_config):
+        service = Mock()
+        pyramid_config.register_service(service, name='user')
+        return service
+
 
-@pytest.mark.usefixtures('group_service', 'links_service')
+@pytest.mark.usefixtures('group_service', 'links_service', 'user_service')
 class TestAnnotationResource(object):
-    def test_links(self, group_service, links_service):
+    def test_links(self, group_service, links_service, user_service):
         ann = Mock()
-        res = AnnotationResource(ann, group_service, links_service)
+        res = AnnotationResource(ann, group_service, links_service, user_service)
 
         result = res.links
 
         links_service.get_all.assert_called_once_with(ann)
         assert result == links_service.get_all.return_value
 
-    def test_link(self, group_service, links_service):
+    def test_link(self, group_service, links_service, user_service):
         ann = Mock()
-        res = AnnotationResource(ann, group_service, links_service)
+        res = AnnotationResource(ann, group_service, links_service, user_service)
 
         result = res.link('json')
 
         links_service.get.assert_called_once_with(ann, 'json')
         assert result == links_service.get.return_value
 
-    def test_acl_private(self, factories, group_service, links_service):
+    def test_acl_private(self, factories, group_service, links_service, user_service):
```

Comment:
Hmm, there is quite a lot of repeated plumbing here. This is part of the reason why I wonder about adding a `user` property to `AnnotationResource` vs generating `user_info` with a formatter.